src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -1624,24 +1638,32 @@ private IEnumerable<PSModuleInfo> CreateFakeModuleObject(IEnumerable<ModuleSpeci
                     invalidOperation.SetErrorId("Modules_WildCardNotAllowedInModuleToProcessAndInNestedModules");
                     throw invalidOperation;
                 }
+
                 // See if this module is already loaded. Since the manifest entry may not
                 // have an extension and the module table is indexed by full names, we
                 // may have search through all the extensions.
                 PSModuleInfo loadedModule = null;
-                string rootedPath = this.FixupFileName(moduleBase, actualRootModule, extension: null, importingModule);
-                string mtpExtension = Path.GetExtension(rootedPath);
-                if (!string.IsNullOrEmpty(mtpExtension) && ModuleIntrinsics.IsPowerShellModuleExtension(mtpExtension))
+                string rootedPath = null;
+
+                // For a root module, we use its own module name instead of the manifest module name when calling 'FixFileName'.
```

Comment:
This was really confusing to me (why would a module path ever use a parent module name?), until I discovered that `rootedPath` is *only* used as a look up key from the module table.  Please mention this in the comments (or use a more descriptive name than `rootedPath`, like `moduleTableKey`).  

Also this method is waaay to long.

Reply:
This comment is for explaining why `rootModuleName` is passed into `FixFileName` (instead of the name we got from the .psd1 file path).

A root module could be a binary module, and when that's the case, resolving it may require loading the assembly (this is the first loading request, for resolving the path of a binary root module). The `FixFileName` has the `moduleName` parameter's purpose documented.